### ISSUE TITLE
How to harmonize output

### ISSUE THREAD

**Issue by SilasK** [linnabrown/run_dbcan#106]
dbcan perform 3 annotations: Diamond, eCAMI, and HMMR. The output of the tree don't necessarily agree. Do you have a preferred way to harmonize them? 
Should we trust HMMR more? Is DIAMOND really reliable?

I know in the publication you had different comparisons, but this was before eCAMI was added.

**Comment by SilasK** [comment]
https://twitter.com/SilasKieser/status/1574817427208245249

**Comment by yinlabniu** [comment]
In most cases, they agree. For those not, we do have an internal way to harmonize. The preferred order is HMMER, eCAMI, Diamond, meaning if HMMER produces output, we take its annotation over the other two; if HMMER does not have output, we take eCAMI annotation. In this order, we basically ignore the Diamond annotation. However, we do need Diamond for the 2/3 voting. Yes, this was done on Hotpep before eCAMI came out. But the prioritization strategy is inherited.

Yanbin

________________________________
From: Silas Kieser ***@***.***>
Sent: Wednesday, September 28, 2022 2:13 AM
To: linnabrown/run_dbcan ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [linnabrown/run_dbcan] How to harmonize output (Issue #106)

Non-NU Email
________________________________

dbcan perform 3 annotations: Diamond, eCAMI, and HMMR. The output of the tree don't necessarily agree. Do you have a preferred way to harmonize them?
Should we trust HMMR more? Is DIAMOND really reliable?

I know in the publication you had different comparisons, but this was before eCAMI was added.

**Comment by SilasK** [comment]
Thank you for the clarification. Would you still keep diamond prediction for genes that are not annotated with the other two tools? And if yes to down to the default E-value threshold? 

Also if Hmmer predicts one annotation and DIamond 2 you wouldn't keep the second from diamond.

**Comment by yinlabniu** [comment]
No, by default, we remove predictions resulted from just one tool (including those only predicted by diamond). If hmmer and diamond have different annotation, we keep hmmer's.
________________________________

**Comment by yinlabniu** [comment]
that's right. if only hmmer generates output and the other two are not, we drop it. But users can decide to keep it since you have the raw results.

**Comment by Xinpeng021001** [comment]
We'll update it in few days
